### PR TITLE
fix lpmode set failure issue 

### PR DESCRIPTION
**Pull request by keboliu**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 1]
```diff
@@ -18,6 +18,14 @@
 PMAOS_ENABLE = 1
 PMAOS_DISABLE = 2
 
+PORT_TYPE_NVE = 8
+PORT_TYPE_OFFSET = 28
+PORT_TYPE_MASK = 0xF0000000
+
+def check_nve(port):
```
Comment (andriymoroz-mlnx) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: if this function checks whether port is NVE, I suggest to rename it to is_nve()
also it seems to return integer but expected to return boolean
suggest to return (port_type & PORT_TYPE_NVE != 0)
or port_type & PORT_TYPE_NVE == PORT_TYPE_NVE

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 2]
```diff
@@ -18,6 +18,14 @@
...
+    port_type = (port & PORT_TYPE_MASK) >> PORT_TYPE_OFFSET
```
Comment (andriymoroz-mlnx) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: small optimization - NVE mask can be calculated once
NVE_MASK = PORT_TYPE_MASK & (PORT_TYPE_NVE << PORT_TYPE_OFFSET)
then function becomes very simple:
```
def check_nve(port):
    return (port & NVE_MASK) != 0
```
and can be eliminated :)
but this is up to you

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 3]
```diff
@@ -48,6 +56,9 @@ def get_log_ports(handle, sfp_module):
     log_port_list = []
     for i in range(0, port_cnt):
         port_attributes = sx_port_attributes_t_arr_getitem(port_attributes_list, i)
+        is_nve = check_nve(int(port_attributes.log_port))
```
Comment (andriymoroz-mlnx) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: three conditions to check before adding port to the list can be done in one if
e.g. 
if A & B & C:
    list.append(port)

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 4]
```diff
@@ -48,6 +56,9 @@ def get_log_ports(handle, sfp_module):
...
+        if is_nve:
+            continue
         if port_attributes.port_mapping.module_port == sfp_module:
             if get_port_admin_status_by_log_port(port_attributes.log_port):
```
Comment (andriymoroz-mlnx) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: not current change, but if possible, make this "if" more readable
it is not clear what this function returns and when condition is true

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py (hunk 1 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: revised.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py (hunk 2 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: great suggestion, revised!

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py (hunk 3 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: revised

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py (hunk 4 above)
Comment (keboliu) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: renamed the function name to be more meaningful

**Pull request closed without merging**